pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/BaseClusterIntegrationTest.java

```
@@ -16,1402 +16,322 @@
         try {
-          int tableCount = input.length();
-
-          for (int i = 0; i < tableCount; i++) {
-            JSONObject tableRouting = input.getJSONObject(i);
-            String tableName = tableRouting.getString("tableName");
-            if (tableName.startsWith(getTableName())) {
-              JSONArray routingTableEntries = tableRouting.getJSONArray("routingTableEntries");
-              int routingTableEntryCount = routingTableEntries.length();
-              for (int j = 0; j < routingTableEntryCount; j++) {
-                JSONObject routingTableEntry = routingTableEntries.getJSONObject(j);
-                Iterator<String> routingTableEntryInstances = routingTableEntry.keys();
-
-                // The randomly selected instance should not be in the routing table
-                while (routingTableEntryInstances.hasNext()) {
-                  String instance = routingTableEntryInstances.next();
-                  if (instance.equals(instanceAddress)) {
-                    return false;
-                  }
-                }
+          ExternalView externalView = _helixAdmin.getResourceExternalView(_clusterName, tableNameWithType);
+          Set<String> partitionSet = externalView.getPartitionSet();
+          if (partitionSet.size() == expectedNumSegments) {
+            int numOnlinePartitions = 0;
+            for (String partitionId : partitionSet) {
+              Map<String, String> partitionStateMap = externalView.getStateMap(partitionId);
+              if (partitionStateMap.containsValue("ONLINE")) {
+                numOnlinePartitions++;
               }
             }
+            if (numOnlinePartitions == expectedNumSegments) {
+              return true;
+            }
           }
-
-          return true;
-        } catch (JSONException e) {
-          LOGGER.warn("Caught exception while reading the routing table, will retry", e);
           return false;
+        } catch (Exception e) {
+          return null;
```

Shouldn't we rethrow the exception here?